File: 242-valid-anagram/242-valid-anagram.py
from collections import OrderedDict

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        hm1 = {}
        hm2 = {}
        
        # get count of letters in first string
        for cs in s:
            if cs in hm1:
                hm1[cs] += 1
            else:
                hm1[cs] = 1
        
        # get count of letters in second string   
        for ct in t:
            if ct in hm2:
                hm2[ct] += 1
            else:
                hm2[ct] = 1
        
        # compare hash maps
        if hm1 == hm2:
            return True
        
        return False

File: 242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        // unequal strings
        if (s.size() != t.size()) {
            return false;
        }
        
        // store the element and count in string
        map<char, int> a;
        map<char, int> b;
        
        // populate corresponding maps
        for (auto& i: s) {
            if (a.find(i) == a.end()) {
                a.insert(pair<char, int>(i, 1));
            }
            else a[i]++;
        }
        
        for (auto& j: t) {
            if (b.find(j) == b.end()) {
                b.insert(pair<char, int>(j, 1));
            }
            else b[j]++;
        }
        
        if (a == b) {
            return true;
        }
        
        return false;
    }
};